susemanager-utils/susemanager-sls/src/beacons/pkgset.py

```diff
@@ -1,44 +1,32 @@
 # -*- coding: utf-8 -*-
-'''
+"""
 Watch RPM or DPkg database via cookies and fire
 an event to the SUSE Manager if that has been changed.
 
 Author: Bo Maryniuk <bo@suse.de>
-'''
+"""
 
 from __future__ import absolute_import
-from salt.utils.odict import OrderedDict
 import os
-import logging
-log = logging.getLogger(__name__)
 
 
-__virtualname__ = 'pkgset'
+__virtualname__ = "pkgset"
 
-PKG_PLUGINS = OrderedDict([
-        ("/usr/lib/zypp/plugins/commit/venv-zyppnotify", "/var/cache/venv-salt-minion/rpmdb.cookie"),
-        ("/usr/share/yum-plugins/venv-yumnotify.py",     "/var/cache/venv-salt-minion/rpmdb.cookie"),
-        ("/usr/bin/venv-dpkgnotify",                     "/var/cache/venv-salt-minion/dpkg.cookie"),
-        ("/usr/lib/zypp/plugins/commit/zyppnotify",      "/var/cache/salt/minion/rpmdb.cookie"),
-        ("/usr/share/yum-plugins/yumnotify.py",          "/var/cache/salt/minion/rpmdb.cookie"),
-        ("/usr/bin/dpkgnotify",                          "/var/cache/salt/minion/dpkg.cookie")
-    ])
+PKGSET_COOKIES = (
+    "/var/cache/venv-salt-minion/rpmdb.cookie",
+    "/var/cache/venv-salt-minion/dpkg.cookie",
+    "/var/cache/salt/minion/rpmdb.cookie",
+    "/var/cache/salt/minion/dpkg.cookie",
+)
 COOKIE_PATH = None
 
 
 def __virtual__():
-    global COOKIE_PATH
-
-    for plug, cookie in PKG_PLUGINS.items():
-        if os.path.exists(plug):
-            COOKIE_PATH = cookie
-            break
-
-    return COOKIE_PATH and __virtualname__ or False
+    return __virtualname__
```

Comment:
I think it still makes sense to only load the plugin if it can work. What about checking for the cookie files here? We could use `COOKIE_PATH` to store the correct path and not iterate over the possible locations in `beacon`. Or, at least, do a one time check that at least one cookie exists, e.g. `any(os.path.exists(cookie) for cookie in PKGSET_COOKIES)`

Reply:
The problem is that the cookie file could not exists yet on loading the beacon it could appear some time after on the moment of modifying installed packages on the system. If we stop loading the beacon with such check we will loose all of the packages notifications until next restart of salt-minion

Reply:
Makes sense :+1: